Monika After Story/game/zz_windowreacts.rpy

```diff
@@ -0,0 +1,315 @@
+    ]
+
+init python:
+    import os
+    #The initial setup
+
+    #We can only do this on windows
+    if renpy.windows:
+        #We need to extend the sys path to see our packages
+        import sys
+        sys.path.append(renpy.config.gamedir + '\\python-packages\\')
+
+        #We try/catch/except to make sure the game can run if load fails here
+        try:
+            #Going to import win32gui for use in destroying notifs
+            import win32gui
+            #Import win32api so we know if we can or cannot use notifs
+            import win32api
+
+        except ImportError:
+            #If we fail to import, then we're going to have to make sure nothing can run.
+            store.mas_windowreacts.can_show_notifs = False
+
+            #Log this
+            store.mas_utils.writelog(
+                "[WARNING]: win32api/win32gui failed to be imported, disabling notifications.\n"
+            )
+
+        else:
+            import balloontip
+
+            #Now we initialize the notification class
+            try:
+                tip = balloontip.WindowsBalloonTip()
+
+            except:
+                #So reload still works
+                tip.classAtom = win32gui.UnregisterClass(tip.classAtom, tip.hinst)
+                tip = balloontip.WindowsBalloonTip()
+
+            #Now we set the hwnd of this temporarily
+            tip.hwnd = None
+
+    #List of notif quips (used for topic alerts)
+    notif_quips = [
+        persistent.playername + ", I want to talk to you about something.",
+        "Are you there, " + persistent.playername + "?",
+        "Can you come here for a second?",
+        persistent.playername + ", do you have a second?",
+        "I have something to tell you, " + persistent.playername + "!",
+    ]
+
+    #List of hwnd IDs to destroy
+    destroy_list = list()
+
+    #START: Utility methods
+    def mas_getActiveWindow():
+        if (
+                renpy.windows
+                and mas_windowreacts.can_show_notifs
+                and persistent._mas_windowreacts_windowreacts_enabled
+            ):
+            from win32gui import GetWindowText, GetForegroundWindow
+            return GetWindowText(GetForegroundWindow()).lower()
+        else:
+            #TODO: Mac vers (if possible)
+            #NOTE: Return our process name so we don't show extra notifs because windowreacts disabled
+            return "monika after story"
+
+    def mas_isFocused():
+        """
+        Checks if MAS is the focused window
+        """
+        #TODO: Mac vers (if possible)
+        return store.mas_windowreacts.can_show_notifs and mas_getActiveWindow() == config.name.lower()
+
+    def mas_isInActiveWindow(keywords):
+        """
+        Checks if ALL keywords are in the active window name
+
+        IN:
+            List of keywords
+        """
+        return store.mas_windowreacts.can_show_notifs and len([s for s in keywords if s.lower() not in mas_getActiveWindow()]) == 0
```

Comment:
because you're reusing the results from `mas_getActiveWindow`, should just store those results in a tempvar and then use them in the loop